src/dataflow/source/kafka.rs

```diff
@@ -134,29 +167,36 @@ fn refresh_kafka_metadata(
 fn create_kafka_consumer(
     activator: Arc<Mutex<SyncActivator>>,
     topic: &str,
+    partition_id: i32,
     kafka_config: &ClientConfig,
 ) -> BaseConsumer<GlueConsumerContext> {
     let cx = GlueConsumerContext(activator);
     let consumer: BaseConsumer<GlueConsumerContext> = kafka_config
         .create_with_context(cx)
         .expect("Failed to create Kafka Consumer");
-    consumer.subscribe(&[topic]).unwrap();
+    let mut partition_list = TopicPartitionList::with_capacity(1);
+    partition_list.add_partition(topic, partition_id);
```

Comment:
Potentially dumb question: why did things work before when we weren't doing this?

Reply:
We were using the auto-rebalancing (round-robin) scheme where Kafka automatically assigns consumers to partitions in a round-robin fashion. However, it occasionally takes a while to do that, and it was causing some of the test-drive tests to occasionally fail.